### PR TITLE
Add Brakeman checks

### PR DESCRIPTION
**Pull request by mafonso**
### Jira link

HOTT-367

### What?

I have added/removed/altered:

- [ ] .gitignored `vendor` path
- [ ] Added Brakeman gem and CI jobs

### Why?

I am doing this because:

- We want to run brakeman checks.

### Deployment risks (optional)

- vendor path was removed from the repo. Cypress tests and unit tests pass, but it's not clear at this time if this was needed at some point or it just slipped into the repo.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Ignore bundler config.
 /.bundle
+vendor/*
```
Comment (mafonso) [inline .gitignore:3]: Not clear if this was needed at some point or it just slipped into the repo.
Unit tests and manual cypress tests still pass after this was removed.

Also we had this path under .cfignore, so it never made it to the PaaS, which build everything from scratch.

**Comment by willfish** [comment]
Also, is it possible to fail on a high brakeman issue, only?

**Comment by mafonso** [comment]
> @mafonso Do we want to fix the brakeman issues separately? Or would you prefer to fix them as part of this branch?

There are two tickets on the backlog to address those.  
Since it might require some code refactoring, it may be best to address them on their own stories and keep the scope of this PR just on the CI runs.

**Comment by mafonso** [comment]
> Also, is it possible to fail on a high brakeman issue, only?

Yes, we can choose have `--no-exit-on-warn` option.

File: .circleci/config.yml [hunk 2]
```diff
@@ -134,10 +134,14 @@ jobs:
...
           format: progress
           label: Inspecting with Rubocop
-      - slack/notify:
-          channel: deployments
-          event: fail
-          template: basic_fail_1
+      - run:
+          name: Inspecting with Brakeman
+          when: always
+          command: 'bundle exec brakeman -o test-results/brakeman/brakeman.junit -o brakeman.html --no-progress --separate-models'
```
Comment (willfish) [inline .circleci/config.yml:140]:
```suggestion
          command: 'bundle exec brakeman -o test-results/brakeman/brakeman.junit -o brakeman.html --no-progress --separate-models --no-exit-on-warn'
```

**Review by willfish — APPROVED** [review]
We're going to fix violations separately

**Pull request merged**